### DISCUSSION TITLE
Some videos from cbc.ca can't be saved.

### DISCUSSION THREAD

**Discussion by tiefschneekamerad** [aclap-dev/video-downloadhelper#178] [Help and Q&A]
### Your question

Most of the videos I tried downloading worked just fine, but now I stumbled across some videos that can't be downloaded.
This is one of them:
https://www.cbc.ca/player/play/2135012931727
Is there something I can do?

### Addon & CoApp version and details

Windows 10
version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\tobi",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by Mattmjs** [comment]
I get detection from that link, I have almost the same configuration as you except I'm on companion app version 2.0.19
Maybe you should try a newer version.

**Reply by tiefschneekamerad** [reply]
the video link is detected, but neither resolution version is being downloaded. ButI'll try a new version.

**Reply by Mattmjs** [reply]
Ok I see, they won't allow it :

Error opening input files: Server returned 403 Forbidden (access denied)

**Reply by Mattmjs** [reply]
But guess what, it works in the regular ffmpeg just not in VDH.

**Reply by GrampaWildWilly** [reply]
With Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.1.1a1 beta, CoApp 2.0.19, I just got a Side Download Error.  I must point out again that I do not work for VDH.  I'm just another user.  But I went to Plan B.  I didn't have to fiddle with this one.  There was nothing useful in the Network Monitor until I clicked play in the player on the web page, then paused it after a second or two.  An HLS master manifest showed up in the Network Monitor.  I didn't have to hunt through json items or anything taxing like that.  I just got an m3u8 file right there.  Easy as pie.

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14343243/ffprobe.txt)
[CBC Skiing mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14343245/CBC.Skiing.mp4.Log.txt)
[CBC Skiing vtt Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14343247/CBC.Skiing.vtt.Log.txt)

That gave these results:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f9af5d54-b1e0-4a1d-a8ac-a6b323df2dbf)

And here it is playing in VLC, with the captions:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/98b78660-1f3a-461c-ac3f-4fdc615df63d)

I have to say the captions looked kind of weird.  First off, they were all capital letters, which is annoying.  Then, there seemed to be a lot of overlap, with subtitles displaying over top of each other.  I suppose I could pump this through SubtitleEdit & see what it does with them but I can't be bothered.  I suspect these captions were machine generated.

In any case, VLC played it just fine.  I didn't sit & watch it, just sampled it at approximately 1-minute intervals.  Then I let the last 20 seconds or so play.  I had video, audio, & subtitles right to the end.

I have to say I was prepared to get geoblocked on this.  I'm in the US & I have been geoblocked on other CBC content.  But this one let me in.  Woohoo.

My results tell me that VDH's failure on this content is completely baffling.  There is nothing at all weird about this HLS master manifest.  From my perspective, this should have presented no problem to VDH.  We'll have to see what @paulrouget has to say about this.

**Reply by GrampaWildWilly** [reply]
PS: I used the ffprobe & ffmpeg supplied with CoApp 2.0.19 in my test.  I've made it a policy to use the ffmpeg from the CoApp in all my downloads, including those having nothing at all to do with this forum.

**Reply by GrampaWildWilly** [reply]
PPS:  I didn't look carefully before.  Now I've looked again.  The VDH error indeed is a 403 Forbidden.  That is completely weird.  Why would VDH get 403 but good old-fashioned Plan B have no problems?  That is beyond my knowledge.

**Comment by tiefschneekamerad** [comment]
Wow!!! Thank you very much @Mattmjs  and @GrampaWildWilly !!!
Unfortunately I have no idea on how to use ffmpeg in order to donwload the video. Can either of you please point me in the right direction?

**Reply by GrampaWildWilly** [reply]
Just study the text files I posted in my response above.

**Reply by GrampaWildWilly** [reply]
Maybe there is one missing link.  Do this in this order.

1. Copy the URL of the web page to your clipboard.
2. Open an empty browser window.
3. Open the Network Monitor, as shown in the image below.
4. Paste the URL from step 1 into the URL bar of the empty browser window.
5. Hit Enter.
6. Click play in the video player on the web page.  Keep the Network Monitor open throughout.
7. Pause playback after a second or two.
8. Scroll through the Network Monitor until you find the indicated entry.  It helps to sort on Type, as I've indicated in the image below.  That entry is the HLS master manifest.
9. Grab the URL of the HLS master manifest as I've shown below.  Copying the URL puts it in your clipboard.

That should give you enough to make the study of my text files worthwhile.

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0b10d17e-8136-46ff-946f-d7fd8770c39a)

**Comment by tiefschneekamerad** [comment]
Perfect! Perfect! PERFECT!!! Thank you very much!!!

**Reply by tiefschneekamerad** [reply]
hmmm @GrampaWildWilly I do get a playlist with mp4 files of the same video, in different resolutions, but niehter of them has audio?

**Reply by GrampaWildWilly** [reply]
Did you get the URL of the HLS master manifest?  That's the one whose name has master in it & is of file type m3u8.  Did you pass that to ffprobe?  What did you get?  Post the ffprobe report file here.  That would be:

`ffprobe "https://blah blah master.m3u8 blah blah" 2>"ffrpobe capture file.txt"`

The quotation marks are critical.  Be sure you code them.  Also the spacing around 2> is required like I've written it here.  My log files showing the execution of ffmpeg end with invocations of ffprobe showing exactly how I executed it.  Then post that capture file here.  The exact name of the capture file can be anything you choose.  On any partition.  In any directory.  With any file name.  Just make it a .txt file.

**Reply by tiefschneekamerad** [reply]
my bad, it was the wrong link... it worked like a charm!!! Thanks again! Very very much!!!

**Reply by tiefschneekamerad** [reply]
now it didn't work with a second video... https://www.cbc.ca/player/play/2133806147684
[Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14347664/Log.txt)

**Reply by GrampaWildWilly** [reply]
When I click play on that one, I get this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ad3c6b7a-ec9e-48b4-9ed6-190cc6cb08a7)

Maybe I'm being geoblocked on this one.

In your log, it shows 403 Forbidden.  That is probably the source of my failure & perhaps yours to even play the content.  Oh.  Wait.  You have multiple attempts to invoke ffmpeg in there.  Never mind.

In addition, it looks like you coded `-map 0:7`.  There is no such stream in the video you have managed to detect.  You ALWAYS need to execute ffprobe & LOOK AT the streams that are being offered.  You have to look at the numbers ffprobe tells you, like 0:0, 0:1, 0:2, etc.  Then you pick the video track & the audio track you want to download & put those numbers into the -map parameters on ffmpeg.  NEVER skip ffprobe.  In your log, it looks like you want
`-map 0:5 -map 0:6`.  There is no stream 0:7.

Also, if you are not interested in captions, you can just remove `-strict experimental` from both ffprobe & ffmpeg.  If subtitles are present, without the -strict parameter, you'll just see an error message about it.  If you then decide you actually want the captions, put the -strict parameter back onto your invocations of the the 2 programs.
